eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/BomGenerator.java

```diff
@@ -0,0 +1,198 @@
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.apache.maven.model.io.xpp3.MavenXpp3Writer;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+
+import static com.azure.tools.bomgenerator.Utils.AZURE_CORE_GROUPID;
+import static com.azure.tools.bomgenerator.Utils.AZURE_CORE_TEST_LIBRARY;
+import static com.azure.tools.bomgenerator.Utils.AZURE_PERF_LIBRARY_IDENTIFIER;
+import static com.azure.tools.bomgenerator.Utils.AZURE_TEST_LIBRARY_IDENTIFIER;
+import static com.azure.tools.bomgenerator.Utils.EXCLUSION_LIST;
+import static com.azure.tools.bomgenerator.Utils.POM_TYPE;
+import static com.azure.tools.bomgenerator.Utils.SDK_DEPENDENCY_PATTERN;
+import static com.azure.tools.bomgenerator.Utils.SDK_NON_GA_PATTERN;
+
+public class BomGenerator {
+    private String outputFileName;
+    private String inputFileName;
+    private String pomFileName;
+    private String externalDependenciesFileName;
+
+    private static Logger logger = LoggerFactory.getLogger(BomGenerator.class);
+
+    BomGenerator() {
+    }
+
+    public void setInputFile(String inputFileName) {
+        this.inputFileName = inputFileName;
+    }
+
+    public void setOutputFile(String outputFileName) {
+        this.outputFileName = outputFileName;
+    }
+
+    public void setPomFile(String pomFileName) {
+        this.pomFileName = pomFileName;
+    }
+
+    public void setExternalDependenciesFile(String externalDependenciesFileName) {
+        this.externalDependenciesFileName = externalDependenciesFileName;
+    }
+
+
+    public void generate() {
+        TreeSet<BomDependency> inputDependencies = scan();
+        TreeSet<BomDependency> externalDependencies = resolveExternalDependencies();
+
+        // 1. Create the initial tree and reduce conflicts.
+        // 2. And pick only those dependencies. that were in the input set, since they become the new roots of n-ary tree.
+        DependencyAnalyzer analyzer = new DependencyAnalyzer(inputDependencies, externalDependencies);
+        TreeSet<BomDependency> outputDependencies = analyzer.analyze();
+        outputDependencies.retainAll(inputDependencies);
+
+        // 2. Create the new tree for the BOM.
+        analyzer = new DependencyAnalyzer(outputDependencies, externalDependencies);
+        outputDependencies = analyzer.analyze();
+
+        // 3. Validate that the current BOM has no conflicts.
+        boolean validationPassed = analyzer.validate();
+
+        // 4. Create the new BOM file.
+        if(validationPassed) {
+            rewriteBomFile();
+            writeBom(outputDependencies);
```

Comment:
Yeah this is just a convenience added to make the life easier for the reviewer of the BOM for the first time the tool is run. The tool writes the BOM in a particular order, so the first time the tool is run I rewrite the existing POM to also have that order so the diff is easier - I will be removing this code in the later iteration (or it will automatically become no-op after the fact).

Reply:
Added a comment for it.